### DISCUSSION TITLE
Remove dont work

### DISCUSSION THREAD

**Discussion by Euphillya** [TheElectronWill/night-config#158] [Q&A] [answered]
Hello, I don't know why this is happening to me but remove() doesn't work, it doesn't remove the configuration, is this normal?

My code :
https://github.com/Euphillya/Skyllia/blob/2129469c24486233f1b51dd2d613aa9992bd589d/plugin/src/main/java/fr/euphyllia/skyllia/configuration/ConfigToml.java#L82

**Comment by TheElectronWill** [comment]
Hello!

`remove` deletes an entry from the config, for example if the config is (toml syntax for the example):
```toml
a = "aa"
b = "bb"
```
you can `config.remove("a")` and only `b` will remain in your config (Well, in the *in-memory version* of the config, the file is unchanged until you `config.save()` or `writer.write(config, ...)`, unless you enabled autosave ^^).

Can you show me how you call `remove` (your function)? What do you expect it to do?

**Comment by Euphillya** [comment]
https://github.com/Euphillya/Skyllia/blob/2129469c24486233f1b51dd2d613aa9992bd589d/plugin/src/main/java/fr/euphyllia/skyllia/configuration/ConfigToml.java#L191

Autosave is used : 

https://github.com/Euphillya/Skyllia/blob/2129469c24486233f1b51dd2d613aa9992bd589d/plugin/src/main/java/fr/euphyllia/skyllia/configuration/ConfigToml.java#L45

**Comment by TheElectronWill** [comment]
Hmm 🤔 Okay, maybe something else is overwriting the removed value? Or maybe it's the wrong path? You're not using the config from multiple threads, are you?

It's probably not a bug in `config.remove` itself, but I'm sure you'll figure it out after some more debugging :)

**Comment by Euphillya** [comment, marked as answer]
It's okay, I understand the problem... I just wasn't doing the right version check. My apologies.

**Comment by TheElectronWill** [comment]
No worries, have fun programming!
